File: ontrack-model/src/test/java/net/nemerosa/ontrack/model/security/AccountTest.java
package net.nemerosa.ontrack.model.security;

import org.apache.commons.lang3.SerializationUtils;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class AccountTest {

    @Test
    public void serializable_account() {
        Account account = baseAccount();
        // Serialisation
        byte[] bytes = SerializationUtils.serialize(account);
        // Deserialisation
        Account readAccount = SerializationUtils.deserialize(bytes);
        // Check
        assertEquals(account, readAccount);
    }

    private Account baseAccount() {
        return Account.of("test", "Test", "test@test.com", SecurityRole.USER, AuthenticationSource.none(), false, false);
    }

}

File: ontrack-model/src/test/java/net/nemerosa/ontrack/model/security/AccountTest.kt
package net.nemerosa.ontrack.model.security

import net.nemerosa.ontrack.model.security.Account.Companion.of
import org.apache.commons.lang3.SerializationUtils
import org.junit.jupiter.api.Test
import kotlin.test.assertEquals

class AccountTest {

    @Test
    fun serializable_account() {
        val account = baseAccount()
        // Serialisation
        val bytes = SerializationUtils.serialize(account)
        // Deserialisation
        val readAccount = SerializationUtils.deserialize<Account>(bytes)
        // Check
        assertEquals(account, readAccount)
    }

    private fun baseAccount(): Account {
        return of("Test", "test@test.com", SecurityRole.USER)
    }
}
